libsupport/include/katana/EntityTypeManager.h

```diff
@@ -19,13 +22,76 @@ namespace katana {
 /// EntityTypeID for nodes is distinct from EntityTypeID for edges
 /// This type may either be an atomic type or an intersection of atomic types
 /// EntityTypeID is represented using 8 bits
-using EntityTypeID = uint8_t;
+using EntityTypeID = uint16_t;
 static constexpr EntityTypeID kUnknownEntityType = EntityTypeID{0};
 static constexpr EntityTypeID kInvalidEntityType =
     std::numeric_limits<EntityTypeID>::max();
-/// A set of EntityTypeIDs
-using SetOfEntityTypeIDs =
-    std::bitset<std::numeric_limits<EntityTypeID>::max() + 1>;
+
+/// The minimum size of the dynamically sized SetOfEntityTypeIDs
+static constexpr size_t kDefaultSetOfEntityTypeIDsSize = 256;
+/// The maximum size of the dynamically sized SetOfEntityTypeIDs
+static constexpr size_t kMaxSetOfEntityTypeIDsSize = kInvalidEntityType + 1;
+
+/// A dynamically sized set of EntityTypeIDs
+//TODO(emcginnis): had to use std::vector<bool> since boost::dynamic_bitset is not serializable
+// replace this with katana::DynamicBitset when it is available to libsupport
+// until that is done, bitwise operators and methods like "all" will be very inefficient
+class KATANA_EXPORT SetOfEntityTypeIDs : public std::vector<bool> {
+public:
+  SetOfEntityTypeIDs() { resize(kDefaultSetOfEntityTypeIDsSize); }
+
+  SetOfEntityTypeIDs(size_t size) { resize(size); }
+
+  void set(size_t index) { at(index) = true; }
+
+  void set(size_t index, bool val) { at(index) = val; }
+
+  void set() {
+    for (size_t i = 0; i < size(); i++) {
+      at(i) = true;
+    }
+  }
+
+  void reset() {
+    for (size_t i = 0; i < size(); i++) {
+      at(i) = false;
+    }
+  }
+
+  bool test(size_t index) const { return (at(index) == true); }
+
+  bool all() const {
+    for (auto it : *this) {
+      if (!it) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  // these operators are super inefficient
```

Comment:
Even the above function. Please add this as doxygen to all inefficient functions. 

nit: It's better to not use operator overloading for inefficient functions.